lib/chef/data_collector.rb

```diff
@@ -226,7 +225,7 @@ def send_to_output_locations(message)
       # @param message [Hash] the message to render as JSON
       #
       def send_to_file_location(file_name, message)
-        File.open(file_name, "a") do |fh|
+        File.open(File.expand_path(file_name), "a") do |fh|
```

Comment:
File.open accepts only an absolute file path. Consider the relative's path to avoids error.